tapa/core.py

```diff
@@ -1513,6 +1509,80 @@ def get_aie_graph(self, task: Task) -> str:
             connect_def="\n\t\t".join(connect_def),
         )
 
+    def _visit_tasks(
+        self,
+        target_task: str,
+        current_task: str,
+        current_inst: str,
+        current_hierarchy: list[str],
+        found_hierarchies: list[tuple[str, ...]],
```

Comment:
Suggest yielding or returning `found_hierarchies` instead of appending to an output argument.